src/pykeen/triples/triples_factory.py

```diff
@@ -456,6 +422,29 @@ def from_path(
             compact_id=compact_id,
         )
 
+    def with_other_triples(
```

Comment:
this is no doubt a good function, but needs a better name ;)

Reply:
Suggestions? I first had `replace_triples` but that sounded like an in-place operation.